keps/sig-node/753-sidecar-containers/README.md

```diff
@@ -192,584 +173,1331 @@ logs are lost.
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
 
 ### Upgrade / Downgrade Strategy
-When upgrading no changes should be needed to maintain existing behaviour as all of this behaviour is optional and disabled by default.
-To activate the feature they will need to enable the feature gate and mark their containers as sidecars in the container spec.
 
-When downgrading `kubectl`, users will need to remove the sidecar field from any of their Kubernetes manifest files as `kubectl` will refuse to apply manifests with an unknown field (unless you use `--validate=false`).
+#### Upgrade strategy
+
+Existing sidecars (implemented as regular containers) will still work as
+intended, as in the past we don't recognize the new field `restartPolicy` today.
+
+Upgrade will not change any current behaviors.
+
+#### Downgrade strategy
+
+First, there will be no effect on any workload that doesn't use a new field. Any
+combination of feature gate enabled/disabled or version skew will work as usual
+for that workload.
+
+Due to the new field added to `initContainers` to turn them into sidecars,
+downgrading to the version without this feature will not be possible -  the Pod
+spec will be rejected by the control plane and all kubelets during
+unmarshalling. Workloads will have to be deleted and recreated with the old way
+of handling sidecars.
```

Comment:
Imho, planning out the release milestones for alpha/beta/GA with [supported version skews](https://kubernetes.io/releases/version-skew-policy/) in mind can help avoid ugly back-compat glue at the cost of beta/GA coming a version later than otherwise possible.

I also found it useful to add a [version skew table](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#version-skew-strategy) (mainly to help myself as a visual thinker) to the in-place resize KEP. Please feel free to swipe and adapt if you feel it is helpful.